File: ISPMEDIA-ANGULAR/src/app/entities/GrupoConvite.ts
import {User} from "./User";
import {Grupo} from "./Grupo";
import {EstadoConvite} from "./enums/EstadoConvite";

export class GrupoConvite {
  id!: number;
  anfitriao:User=new User();
  convidado: User = new User();
  grupo: Grupo=new Grupo();
  estadoConvite: EstadoConvite = EstadoConvite.PENDENTE;

  constructor() {
  }
}

File: src/main/java/com/ngolajr/ispmedia/entities/GrupoConvite.java
package com.ngolajr.ispmedia.entities;

import com.ngolajr.ispmedia.entities.enums.EstadoConvite;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class GrupoConvite {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    @ManyToOne
    private Utilizador anfitriao;
    @ManyToOne
    private Utilizador convidado;
    @ManyToOne
    private Grupo grupo;
    private EstadoConvite estadoConvite;
}
